### ISSUE TITLE
Proposal for @colyseus/loadtest

### ISSUE THREAD

**Issue by endel** [colyseus/colyseus#430]
The `colyseus-loadtest` tool currently requires the input file to follow its own "DSL", implementing the methods `onJoin`, `onLeave`, `onStateChange`, etc.

**Example** ([See full example](https://github.com/colyseus/colyseus/blob/a404b0fe9113392ed4c1cae7918ccd79aedc0350/packages/loadtest/example/bot.ts#L1-L25))

```typescript
// ...
export function requestJoinOptions (this: Client, i: number) {
    return { requestNumber: i };
}

export function onJoin(this: Room) {
    console.log(this.sessionId, "joined.");

    this.onMessage("*", (type, message) => {
        console.log("onMessage:", type, message);
    });
}

export function onLeave(this: Room) {
    console.log(this.sessionId, "left.");
}
// ...
```

## Proposed behaviour

No "DSL": It would be more intuitive (and less learning curve required) to allow the user to provide a plain script file that simply consumes the regular `colyseus.js` package. By overwriting/monkey-patching `colyseus.js`'s internal methods we can keep tracking bandwidth and memory usage by the client while making it easier to write "bot" scripts.

Multi-threading: as Node.js is single-threaded, the more concurrent connections spawned through the loadtest tool, less responsive the Node.js process will become - making some client connections drop because of that. Depending on the CPU, 300 connections are enough to start losing client connections.

**Example / source-code:**

```typescript
import { Client } from "colyseus.js";

export async function main(options: any) {
  const client = new Client(options.endpoint);
  const room = await client.joinOrCreate(options.room);
  room.onMessage("msg-type", (payload) => {
    // ...
  });
  room.onLeave((code) => {
    // ...
  });
}
```

**Comment by endel** [comment]
Just leaving this as a reference... On `@colyseus/testing` the monkey-patching has been done this way, we can do similarly for `colyseus.js` during the loadtest: 
https://github.com/colyseus/colyseus/blob/8dd5b6407122cdcab362f5f534bad68eef1dda82/packages/testing/src/Room.ext.ts#L29-L35

**Comment by endel** [comment]
Landed on 0.15 https://docs.colyseus.io/tools/loadtest/#example

Multi-threading is still not supported though.
